runtime/near-vm-runner/src/runner.rs

```diff
@@ -108,3 +108,53 @@ impl VMKind {
         }
     }
 }
+
+/// Type returned by any VM instance after loading a contract and executing a
+/// function inside.
+#[derive(Debug, PartialEq)]
+pub enum VMResult {
+    /// Execution was not able to start because preconditions were not met.
+    NotRun(VMError),
+    /// Execution started but hit an error.
+    Aborted(VMOutcome, VMError),
+    /// Execution finished without error.
+    Ok(VMOutcome),
+}
+
+impl VMResult {
+    pub fn not_run(error: VMError) -> VMResult {
+        VMResult::NotRun(error)
+    }
+    #[cfg_attr(not(feature = "protocol_feature_function_call_weight"), allow(unused_mut))]
+    pub fn abort(mut logic: VMLogic, error: VMError) -> VMResult {
+        let outcome = logic.compute_outcome_and_distribute_gas();
+        VMResult::aborted(outcome, error)
+    }
+    pub fn aborted(outcome: VMOutcome, error: VMError) -> VMResult {
+        VMResult::Aborted(outcome, error)
+    }
+    pub fn ok(outcome: VMOutcome) -> VMResult {
+        VMResult::Ok(outcome)
+    }
+    pub fn outcome(&self) -> Option<&VMOutcome> {
+        match self {
+            VMResult::NotRun(_err) => None,
+            VMResult::Aborted(outcome, _err) => Some(outcome),
+            VMResult::Ok(outcome) => Some(outcome),
+        }
+    }
+    pub fn error(&self) -> Option<&VMError> {
+        match self {
+            VMResult::NotRun(err) => Some(err),
+            VMResult::Aborted(_outcome, err) => Some(err),
+            VMResult::Ok(_outcome) => None,
+        }
+    }
+    pub fn outcome_error(self) -> (Option<VMOutcome>, Option<VMError>) {
```

Comment:
Please add some space between- and doc-comments to the methods here.

Reply:
Done